# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxHarfBuzzShaper.h
# Commit: 3b719cd12933
# Full Hash: 3b719cd1293370d843691b3656b52b04042d5ec4
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2022-03-02 04:31:47
# Regressor Bug: 1756720
# File Overlap Count: 1
# Description:
#   Bug 1756720 - When using legacy MS Symbol fonts, map character codes from the U+00xx range to the PUA range U+F0xx found in the cmap subtables. r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D139724
# ==============================================================================

diff -r 9849c6e79fa3 -r 3b719cd12933 gfx/thebes/gfxHarfBuzzShaper.h
--- a/gfx/thebes/gfxHarfBuzzShaper.h	Tue Mar 01 17:27:27 2022 +0200
+++ b/gfx/thebes/gfxHarfBuzzShaper.h	Tue Mar 01 15:22:56 2022 +0000
@@ -169,6 +169,12 @@
   // Whether the font implements GetGlyph, or we should read tables
   // directly
   bool mUseFontGetGlyph;
+
+  // Whether the font is an MS Symbol-encoded font, in which case we will
+  // try remapping U+0020..00FF to U+F020..F0FF for characters in the U+00xx
+  // range that are otherwise unsupported.
+  bool mIsSymbolFont;
+
   // Whether the font implements GetGlyphWidth, or we should read tables
   // directly to get ideal widths
   bool mUseFontGlyphWidths;